source/develop/release-management/features/network/openstack_identity_api_v3.html.md

```diff
@@ -0,0 +1,300 @@
+~~~~
+HTTP/1.1 201 Created
+X-Subject-Token: gAAAAABbH3sSGLgtxvoHyYDLZd6fSuserBHG2P-3TdykaYnq-eYo7aUBdZK0I
+Vary: X-Auth-Token
+Content-Type: application/json
+
+{"token": ... }
+~~~~
+
+### Implementation in oVirt Engine
+
+To support OpenStack Identity API v3 oVirt engine has to be extended to handle
+the user domain name, the project name and the domain name of the project.
+This requires changes in webadmin, REST-API, the database and the 
+ExternalNetworkTokenProvider in the backend.
+
+The version of the OpenStack Identity API is derived from the authentication
+URL. If the authentication URL ends with "/v3" or "/v3/", OpenStack Identity
+API v3 will be used, else OpenStack Identity API v2.0 to ensure backward
+compatibility.
+
+The default ovirt-provider-ovn has not be modified, because the OpenStack
+Identity API v2.0 is not removed.
+
+### Implementation in oVirt Provider OVN
+To support OpenStack Identity API v3 the oVirt Provider OVN to be extended to
+handle requests to `/v3/auth/tokens`. The `scope` can be ignored, like the
+`tenantName`.
+In oVirt Provider OVN's OpenStack Identity API v2.0 implementation the
+user name has to match the following format:
+`<admin_username>[@<fqdn>]@<ovirt_profile>`, while the optional `<fqdn>` is
+the Active Directory or LDAP domain. For the reason of consistency, this format
+is kept in the OpenStack Identity API v3 implementation and the `user.domain`
+is ignored.
+
+## Prerequisites
+
+oVirt Engine requires an updated version of [OpenStack Java SDK 3.2.x][9] to
+accessing OpenStack Identity API v3.
+
+## Limitations
+
+OpenStack Identity API provides a catalog of multiple endpoints, which
+provides the URL to access the OpenStack service. This feature does change
+the current implementation in oVirt Engine, which requires the user to specify
+this URL manually.
+
+## Benefit to oVirt
+
+Support for OpenStack Identity API v3 in oVirt Engine is required to enable 
+access to services from [OpenStack instances which does not support Identity
+API v2.0, especially the OpenStack releases since Qeens.][12]
+
+## User Experience
+
+### Webadmin
+
+The "Add Provider" popup has to provide inputs for the new properties user
+domain name, the project name and the domain name of the project.
+If the authentication URL ends with "/v3" or "/v3/", the user domain name, the
+project name and the domain name of the project will be used, else the tenant
+name will be used.
+
+![add a new provider](../../../../images/features/network/openstack_idendity_api-add-dialog-provider.png)
+
+### Engine-Setup
+
+The ovirt-provider-ovn external provider which is configured by engine-setup is
+not modified and continues to use OpenStack Identity API v2.0, because of the
+Engine and the OVN provider has to support OpenStack Identity API v2.0 for the
+sake of backward compatibility anyway and switching to OpenStack Identity API
+v3 would be no improvement but introduce the risk of new implementation bugs.
+
```

Comment:
It's ok for upgrade flow, but for new installation i don't see any reason to keep the v2 as default one. or change the default or add option to choose.

Reply:
Sure, this would be nice to have, but as long as there are no clients tools which require keystone v3, there is no pressure.